Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp

```diff
@@ -1339,7 +1343,33 @@ static gboolean handleScroll(WebKitWebViewBase* webViewBase, double deltaX, doub
         return GDK_EVENT_PROPAGATE;
 
     auto* event = gtk_event_controller_get_current_event(eventController);
+    if (event && !priv->shouldForwardNextWheelEvent.isEmpty()) {
+        // Some events that were meant for delayed propagation may have been coalesced/compressed by GDK into the current event:
+        // https://docs.gtk.org/gtk4/migrating-3to4.html#adapt-to-gdkevent-api-changes
+        // There's no way to both propagate and stop the current event, so just clear out these obsolete events from the to-be-propagated vector.
+        unsigned length = 0;
+        auto* history = gdk_event_get_history(event, &length);
```

Comment:
Use GUnqiuePtr, something like:

```suggestion
        GUniquePtr<GdkTimeCoord*> history(gdk_event_get_history(event, &length));
```

Reply:
Done